packages/components/src/box-control/input-control.tsx

```diff
@@ -147,51 +164,135 @@ export default function BoxInputControl( {
 	const usedValue =
 		mergedValue === undefined && computedUnit ? computedUnit : mergedValue;
 	const mixedPlaceholder = isMixed || isMixedUnit ? __( 'Mixed' ) : undefined;
+	const hasPresets = presets && presets.length > 0 && presetKey;
+	const hasPresetValue =
+		hasPresets &&
+		mergedValue !== undefined &&
+		! isMixed &&
+		isValuePreset( mergedValue, presetKey );
+	const [ showCustomValueControl, setShowCustomValueControl ] = useState(
+		! hasPresets ||
+			( ! hasPresetValue && ! isMixed && mergedValue !== undefined )
+	);
+	const showRangeControl = true;
+	const presetIndex = hasPresetValue
+		? getPresetIndexFromValue( mergedValue, presetKey, presets )
+		: undefined;
+	const marks = hasPresets
+		? [ { value: 0, label: __( 'None' ) } ].concat(
+				presets.map( ( preset, index ) => ( {
+					value: index + 1,
+					label: preset.name,
```

Comment:
What about falling back to the slug in case there's not a name here?

Reply:
Do we have presets where names are optional?

Reply:
Humans + JSON 😄 Also, they're not marked as required in theme.json.

At least when I working on typography closely, theme.json font size presets weren't guaranteed, and it didn't hurt to have the backup.

Reply:
Sure that seems easy to add anyway :)